packages/core/src/lib/connection_manager.ts

```diff
@@ -115,7 +117,12 @@ export class ConnectionManager {
     }
 
     try {
-      log(`Deleting undialable peer ${peerId.toString()} from peer store`);
+      log(
+        `Deleting undialable peer ${peerId.toString()} from peer store. Error: ${JSON.stringify(
```

Comment:
I would advice to change `lastErrorRecorder` to a map where we have `peerId` - `error` relationship.
Need to make sure to clean it once it is not needed not to increase memory consumption (for now not issue but later it could become one).

Reply:
Agree with comment above. I think it can be cleaned as soon as used.

Is this necessary? If a dev see the errors for this peer id in the logs and then see that it's deleted I think they can correlate easily, especially that peer id is present.

Also, do not JSON/string manipulate errors:
 log( `Deleting undialable peer ${peerId.toString()} from peer store`, lastErrorRecorded)

Reply:
As I see we do only `stringify` on the error, then to me it makes sense to have `Map<peerId, string>`.
Another improvement I would advice is to identify what we need to log from the error - maybe only message property? The reason is that `stringify` can throw if the object has circular dependencies and since we do not own `libp2p` it might be anything.